test/controllers/ImportVATControllerSpec.scala

```diff
@@ -0,0 +1,143 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import base.ControllerSpecBase
+import controllers.Assets.Redirect
+import controllers.actions.FakeDataRetrievalAction
+import forms.ImportVATFormProvider
+import mocks.repositories.MockSessionRepository
+import models.{UnderpaymentAmount, UnderpaymentType, UserAnswers}
+import pages.{ImportVATPage, UnderpaymentTypePage}
+import play.api.http.Status
+import play.api.mvc.{AnyContentAsFormUrlEncoded, Call, Result}
+import play.api.test.FakeRequest
+import play.api.test.Helpers.{charset, contentType, defaultAwaitTimeout, status}
+import views.html.ImportVATView
+
+import scala.concurrent.Future
+
+class ImportVATControllerSpec extends ControllerSpecBase {
+
+  val userAnswersWithUnderpayment: Option[UserAnswers] = Some(UserAnswers("some-cred-id")
+    .set(
+      UnderpaymentTypePage,
+      UnderpaymentType(false, true, true)
+    ).success.value
+  )
+
+  private def fakeRequestGenerator(original: String, amended: String): FakeRequest[AnyContentAsFormUrlEncoded] =
+    fakeRequest.withFormUrlEncodedBody(
+      "original" -> original,
+      "amended" -> amended
+    )
+
+  trait Test extends MockSessionRepository {
+    lazy val controller = new ImportVATController(
+      authenticatedAction,
+      dataRetrievalAction,
+      dataRequiredAction,
+      mockSessionRepository,
+      messagesControllerComponents,
+      importVATView,
+      form
+    )
+    private lazy val importVATView = app.injector.instanceOf[ImportVATView]
+    private lazy val dataRetrievalAction = new FakeDataRetrievalAction(userAnswers)
+    val userAnswers: Option[UserAnswers] = Some(UserAnswers("some-cred-id"))
+    val formProvider: ImportVATFormProvider = injector.instanceOf[ImportVATFormProvider]
+    MockedSessionRepository.set(Future.successful(true))
+    val form: ImportVATFormProvider = formProvider
+  }
+
+  "GET /" when {
+    "return OK" in new Test {
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      status(result) mustBe Status.OK
+    }
+
+    "return HTML" in new Test {
+      override val userAnswers: Option[UserAnswers] = Option(
+        UserAnswers("some-cred-id").set(
+          ImportVATPage,
+          UnderpaymentAmount(BigDecimal("40"), BigDecimal(40))
+        ).success.value
+      )
+      val result: Future[Result] = controller.onLoad(fakeRequest)
+      contentType(result) mustBe Some("text/html")
+      charset(result) mustBe Some("utf-8")
+    }
+
+    "should redirect the back button to Customs Duty page" in new Test {
+      controller.backLink(Some(UnderpaymentType(true, true, true))) mustBe Call("GET", controllers.routes.CustomsDutyController.onLoad().toString)
```

Comment:
The correct way to access the text part of the route is to use the url value rather than toString:
controllers.routes.CustomsDutyController.onLoad().url

Please Note this happens in other files too.

Reply:
changed from .toString to .url on ImportVAT controller and controller spec